src/lib/content/coordinators/GetContentItemV2Impl.ts

```diff
@@ -0,0 +1,67 @@
+import { GetContentItemById } from './GetContentItemById';
+import { GetContentItemByKey } from './GetContentItemByKey';
+import { ContentBody } from '../model/ContentBody';
+import { ContentItem } from '../model/ContentItem';
+import { createContentClient } from '../../client/createContentClient';
+import { ContentMapper } from '../mapper/ContentMapper';
+import { AxiosInstance, AxiosResponse } from 'axios';
+import { encodeQueryString } from '../../utils/Url';
+import { ContentClientConfigV2 } from '../../config/ContentClientConfigV2';
+
+/**
+ * @hidden
+ */
+export class GetContentItemV2Impl
+  implements GetContentItemById, GetContentItemByKey {
+  private readonly contentClient: AxiosInstance;
+
+  constructor(
+    private readonly config: ContentClientConfigV2,
+    private readonly mapper: ContentMapper
+  ) {
+    this.contentClient = createContentClient(
+      this.config,
+      `https://${config.hubName}.cdn.content.amplience.net`
+    );
+  }
+
+  getContentItemByKey<T extends ContentBody>(
+    key: string
+  ): Promise<ContentItem<T>> {
+    return this.getContentItem('key', key);
+  }
+  async getContentItemById<T extends ContentBody>(
+    id: string
+  ): Promise<ContentItem<T>> {
+    return this.getContentItem('id', id);
+  }
+
+  async getContentItem<T extends ContentBody>(
+    requestType: 'id' | 'key',
+    value: string
+  ): Promise<ContentItem<T>> {
+    const args = [
+      ['depth', 'all'],
+      ['format', 'inlined'],
```

Comment:
Originally, in the primordial soup of CD2, low latency was seen as a critical NFR. Our API defaults to format `linked` rather than `inlined`. The former is more efficient, but less useful. Also, with depth: `root` vs `all` - its clearly more efficient to read the root item, but less useful for a content tree.

I think it might be worth having the SDK do `inlined` rather than `linked` as that's the typical use case for most of our SDK users, I think. But, we should support the option for `linked` as well as `inlined`, and also `root` as well as `all`.

Reply:
If you look at the Content Delivery 1 client, it doesn't expose all of the various options that the API supports, I took the same approach here, picked the main use case.

I can expose them as additional options either in this PR or in a follow up PR if this isn't a MUST for now.

Reply:
Yes, we can introduce in a later version.